pandas/core/apply.py

```diff
@@ -307,23 +303,40 @@ def apply_standard(self):
                 # reached via numexpr; fall back to python implementation
                 pass
             else:
-                return self.obj._constructor_sliced(result, index=labels)
+                if reduction_success:
+                    return self.obj._constructor_sliced(result, index=labels)
+                else:
+                    # no exceptions - however reduction was unsuccessful,
+                    # use the computed function result for first element
+                    partial_result = result[0]
+                    if isinstance(partial_result, ABCSeries):
```

Comment:
I would actually remove this as its an api change

Reply:
not averse but that would be a separate change; I think we have an issue below about this

Reply:
This object inference was added as without it `test_replace_method` fails.